model/client.go

```diff
@@ -306,36 +307,54 @@ func (c *Client) doDelete(u string) (*http.Response, error) {
 	return c.httpClient.Do(req)
 }
 
-// Uploads the file specified as an argument to S3 via the AWAT
-func (c *Client) UploadArchiveForTranslation(filename string) (*http.Response, error) {
+// UploadArchiveForTranslation uploads the file specified as an argument to S3 via the AWAT
+func (c *Client) UploadArchiveForTranslation(filename string) (string, error) {
 	inputFile, err := os.Open(filename)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to read input file %s", filename)
+		return "", errors.Wrapf(err, "failed to read input file %s", filename)
 	}
 
 	req, err := http.NewRequest("POST", c.buildURL("/upload"), inputFile)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create HTTP request")
+		return "", errors.Wrap(err, "failed to create HTTP request")
 	}
+
 	req.Header.Set("Content-Type", "application/octet-stream")
 	stat, err := inputFile.Stat()
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to determine file stats for %s", inputFile.Name())
+		return "", errors.Wrapf(err, "failed to determine file stats for %s", inputFile.Name())
 	}
+
 	size := stat.Size()
 	if size == 0 {
-		return nil, errors.New("provided file appears to be empty")
+		return "", errors.New("provided file appears to be empty")
 	}
+
 	req.ContentLength = size
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to send HTTP request to AWAT")
+		return "", errors.Wrap(err, "failed to send HTTP request to AWAT")
 	}
-	defer resp.Body.Close()
+
+	defer func() {
+		_, _ = io.Copy(io.Discard, resp.Body)
+		_ = resp.Body.Close()
+	}()
+
 	if resp.StatusCode != http.StatusAccepted {
-		return resp, errors.Errorf("received unexpected code %d from AWAT", resp.StatusCode)
+		return "", errors.Errorf("received unexpected code %d from AWAT", resp.StatusCode)
 	}
-	return resp, err
+
+	archiveBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", errors.New("failed to read response body")
+	}
+
+	if archiveBytes == nil {
+		return "", errors.New("invalid response data")
+	}
+
+	return string(archiveBytes), nil
```

Comment:
nit: if this is archive name (not contents) I would rename the variable to `archiveNameBytes` or something similar. I got confused between this name and the `archiveID` above.

Reply:
Thank you.  It was really confusing. Fixed